src/DeploymentTemplate.ts

```diff
@@ -542,3 +397,53 @@ export class DeploymentTemplate extends DeploymentDocument {
         return this.getDocumentText(spanOfValueInsideString, parentStringToken.span.startIndex);
     }
 }
+
+class StringParseAndScopeAssignmentVisitor extends Json.Visitor {
+    private readonly _jsonStringValueToTleParseResultMap: Map<Json.StringValue, TLE.ParseResult> = new Map<Json.StringValue, TLE.ParseResult>();
+    private readonly _scopeStack: TemplateScope[] = [];
+    private _currentScope: TemplateScope;
+    private readonly _allScopesInTemplate: TemplateScope[] = [];
+
+    public constructor(private readonly _dt: DeploymentTemplate) {
+        super();
+        this._currentScope = _dt.topLevelScope;
+    }
+
+    public createParsedStringMap(): Map<Json.StringValue, TLE.ParseResult> {
```

Comment:
>createParsedStringMap [](start = 11, length = 21)

This is a little weird to me in that if you call it multiple times on the same object, it will keep modifying the object, so you'll get different results, right?